### ISSUE TITLE
SMB copy operation fails due to error in FileVerifier

### ISSUE THREAD

**Issue by dkbz** [arpitkh96/AmazeFileManager#302]
When copying to/from SMB share an error occurs when next file to copy is greatly larger than previous files. It appears that there is a bug in FileVerifier thread implementation. After copying a large file, FileVerifier's add method tries to start again FileVerifier thread while the thread already terminated, because verification of small files were much quicker. Here's trace:

E Copy    : basefile     /storage/emulated/0/bluetooth/Video000.mp4
E Copy    : /storage/emulated/0/bluetooth/Video000.mp4
E Copy    : basefile     /storage/emulated/0/bluetooth/Video001.mp4
E Copy    : /storage/emulated/0/bluetooth/Video001.mp4
W System.err: java.lang.IllegalThreadStateException: Thread already started
W System.err:    at java.lang.Thread.checkNotStarted(Thread.java:849)
W System.err:    at java.lang.Thread.start(Thread.java:1059)
W System.err:    at com.amaze.filemanager.services.FileVerifier.add(FileVerifier.java:54)
W System.err:    at com.amaze.filemanager.services.CopyService$DoInBackground$Copy.copyFiles(CopyService.java:341)
W System.err:    at com.amaze.filemanager.services.CopyService$DoInBackground$Copy.execute(CopyService.java:260)
W System.err:    at com.amaze.filemanager.services.CopyService$DoInBackground.doInBackground(CopyService.java:151)
W System.err:    at com.amaze.filemanager.services.CopyService$DoInBackground.doInBackground(CopyService.java:137)
W System.err:    at android.os.AsyncTask$2.call(AsyncTask.java:295)
W System.err:    at java.util.concurrent.FutureTask.run(FutureTask.java:237)
W System.err:    at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234)
W System.err:    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
W System.err:    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
W System.err:    at java.lang.Thread.run(Thread.java:818)
E Copy    : Got exception checkout

**Comment by floriangosse** [comment]
I got the same error on a Moto G 2nd gen. while copying from internal storage on sd card.

```
09-25 15:17:17.507 19172 19190 W System.err: java.lang.IllegalThreadStateException: Thread already started
09-25 15:17:17.515 19172 19190 W System.err: 	at java.lang.Thread.checkNotStarted(Thread.java:849)
09-25 15:17:17.516 19172 19190 W System.err: 	at java.lang.Thread.start(Thread.java:1059)
09-25 15:17:17.516 19172 19190 W System.err: 	at com.amaze.filemanager.services.FileVerifier.add(FileVerifier.java:54)
09-25 15:17:17.516 19172 19190 W System.err: 	at com.amaze.filemanager.services.CopyService$DoInBackground$Copy.copyFiles(CopyService.java:341)
09-25 15:17:17.516 19172 19190 W System.err: 	at com.amaze.filemanager.services.CopyService$DoInBackground$Copy.copyFiles(CopyService.java:324)
09-25 15:17:17.516 19172 19190 W System.err: 	at com.amaze.filemanager.services.CopyService$DoInBackground$Copy.execute(CopyService.java:260)
09-25 15:17:17.516 19172 19190 W System.err: 	at com.amaze.filemanager.services.CopyService$DoInBackground.doInBackground(CopyService.java:151)
09-25 15:17:17.516 19172 19190 W System.err: 	at com.amaze.filemanager.services.CopyService$DoInBackground.doInBackground(CopyService.java:137)
09-25 15:17:17.516 19172 19190 W System.err: 	at android.os.AsyncTask$2.call(AsyncTask.java:295)
09-25 15:17:17.516 19172 19190 W System.err: 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
09-25 15:17:17.516 19172 19190 W System.err: 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234)
09-25 15:17:17.516 19172 19190 W System.err: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
09-25 15:17:17.516 19172 19190 W System.err: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
09-25 15:17:17.516 19172 19190 W System.err: 	at java.lang.Thread.run(Thread.java:818)
```

**Comment by vishal0071** [comment]
Try latest beta, should be fixed.

**Comment by floriangosse** [comment]
Cool, thanks. I will try it.
I there a plan to release it as a stable version?

**Comment by vishal0071** [comment]
Ofcourse! After fixing a few more critical bugs, it'll make to a stable release. But first test this and report! (:

**Comment by dkbz** [comment]
Hi! Indeed it looks like this bug is fixed in latest beta. I've tried various copy operations and everything successfully completed.

However I've noticed another bug in copy operations - files are being written in 60 kB units. For example, when I copy 4 byte file (ASCII text "TEST") I get 60 kB file (ASCII text "TEST" followed by ~60 k null bytes). It's probably caused by using `out.write(bytes)` instead of `out.write(bytes, 0, length)` in WriteThread.java.
